doc/api/esm.md

```diff
@@ -879,20 +882,20 @@ validating the import assertion.
 
 The final value of `format` must be one of the following:
 
-| `format`     | Description                    | Acceptable types for `source` returned by `load`      |
-| ------------ | ------------------------------ | ----------------------------------------------------- |
-| `'builtin'`  | Load a Node.js builtin module  | Not applicable                                        |
-| `'commonjs'` | Load a Node.js CommonJS module | Not applicable                                        |
-| `'json'`     | Load a JSON file               | { [`string`][], [`ArrayBuffer`][], [`TypedArray`][] } |
-| `'module'`   | Load an ES module              | { [`string`][], [`ArrayBuffer`][], [`TypedArray`][] } |
-| `'wasm'`     | Load a WebAssembly module      | { [`ArrayBuffer`][], [`TypedArray`][] }               |
+| `format`     | Description                    | Acceptable types for `source` returned by `load`                           |
+| ------------ | ------------------------------ | -------------------------------------------------------------------------- |
+| `'builtin'`  | Load a Node.js builtin module  | Not applicable                                                             |
+| `'commonjs'` | Load a Node.js CommonJS module | { [`string`][], [`ArrayBuffer`][], [`TypedArray`][], `null`, `undefined` } |
+| `'json'`     | Load a JSON file               | { [`string`][], [`ArrayBuffer`][], [`TypedArray`][] }                      |
+| `'module'`   | Load an ES module              | { [`string`][], [`ArrayBuffer`][], [`TypedArray`][] }                      |
+| `'wasm'`     | Load a WebAssembly module      | { [`ArrayBuffer`][], [`TypedArray`][] }                                    |
 
 The value of `source` is ignored for type `'builtin'` because currently it is
-not possible to replace the value of a Node.js builtin (core) module. The value
-of `source` is ignored for type `'commonjs'` because the CommonJS module loader
-does not provide a mechanism for the ES module loader to override the
-[CommonJS module return value](#commonjs-namespaces). This limitation might be
-overcome in the future.
+not possible to replace the value of a Node.js builtin (core) module. Unless a
+non-nullish `source` is supplied for type `'commonjs'`, it will be fetched using
+the CommonJS module loader. When non-nullish `source` is supplied for type
+`'commonjs'`, `require` calls from inside the loaded module will call the
+`resolve` and `load` hooks.
```

Comment:
Okay. What _is_ that behavior? 😀

Reply:
`defaultLoad` returns `null` as a source when `format` is `'commonjs'`.

Reply:
> `defaultLoad` returns `null` as a source when `format` is `'commonjs'`.

So if someone is trying to use the return value of `next`, when `next` is `defaultLoad`, would get a `null` source? Shouldn’t they get the actual source content?

Reply:
They'll get a `null` source, they would need to call `readFile(new URL(url))` to opt-in to the new behavior.

Reply:
Okay, how about:


```suggestion
not possible to replace the value of a Node.js builtin (core) module.

For type `'commonjs'`, if `source` is defined, `require` calls from inside the
loaded module will go through the registered `resolve` and `load` hooks.
If `source` is undefined or `null`, it will be handled by the CommonJS
module loader and not processed by any registered hooks. This behavior
for nullish `source` is temporary; in the future, a `source` value that is
undefined or `null` will not be supported.

The Node.js internal `defaultLoad`, which is the value of `next` for the last
loader in the `load` chain, does not return a value for `source`. If one is desired,
loader authors should load it via a call like `readFile(new URL(url))`. This may
also change once support for nullish `source` is discontinued.
```